cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseScanner.java

```diff
@@ -55,13 +63,13 @@ public Row next() {
 
         Map<byte[], byte[]> rowMap = HBaseTable.getRowMap(result, columnFamily);
         if (rowMap.size() > 0) {
-          return new co.cask.cdap.api.dataset.table.Result(result.getRow(), rowMap);
+          byte[] row = rowKeyDistributor == null ? result.getRow() : rowKeyDistributor.getOriginalKey(result.getRow());
```

Comment:
The Rowkey is used by `FactScanner` to get measure name from the row key to create `FactScanResult`